Azure Managed Instance for Apache Cassandra (Cosmos DB)

TheovanKraay opened the pull request.

katbyte commented on internal/services/cosmos/cosmosdb_cassandra_datacenter_resource.go:
```diff
@@ -0,0 +1,251 @@
+			"node_count": {
+				Type:         pluginsdk.TypeInt,
+				Optional:     true,
+				ValidateFunc: validation.IntAtLeast(3),
```
is there a maximum number here?

katbyte reviewed (changes requested):
Thanks for the pr @TheovanKraay - overall looks good aside from some comments i've left line. Additionally we need to  merge main in and fix the merge conflicts. otherwise provided tests pass this should be good to go!

TheovanKraay commented on internal/services/cosmos/cosmosdb_cassandra_datacenter_resource.go:
```diff
@@ -0,0 +1,251 @@
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+			},
+			"disk_capacity": {
```
it is number of P30 disks

TheovanKraay replied:
... to attach to each Cassandra node

TheovanKraay commented:
> Thanks for the pr @TheovanKraay - overall looks good aside from some comments i've left line. Additionally we need to merge main in and fix the merge conflicts. otherwise provided tests pass this should be good to go!

@katbyte - apologies, I had problems with my local branch so I need to abandon this PR. The new PR is https://github.com/hashicorp/terraform-provider-azurerm/pull/14019. I believe I have addressed all your comments there.
